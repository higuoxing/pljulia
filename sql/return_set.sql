CREATE FUNCTION julia_setof_int()
RETURNS SETOF INTEGER AS $$
    for i in 1:5
        return_next(i)
    end
$$ LANGUAGE pljulia;

CREATE FUNCTION julia_setof_array()
RETURNS SETOF INTEGER[] AS $$
x =[[1,2], [3,4]]
for i in x
    return_next(i)
end
$$ LANGUAGE pljulia;

CREATE TYPE named_value AS (
  name   text,
  value  integer
);

CREATE FUNCTION julia_setof_composite()
RETURNS SETOF named_value AS $$
x = [("John", 1), ("Mary", 2)]
for i in x
    return_next(i)
end
$$ LANGUAGE pljulia;

SELECT julia_setof_int();
SELECT julia_setof_array();
SELECT julia_setof_composite();

DROP FUNCTION julia_setof_array();
DROP FUNCTION julia_setof_int();
DROP FUNCTION julia_setof_composite();
DROP TYPE named_value;