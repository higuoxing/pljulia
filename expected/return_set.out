CREATE FUNCTION julia_setof_int()
RETURNS SETOF INTEGER AS $$
    for i in 1:5
        return_next(i)
    end
$$ LANGUAGE pljulia;
CREATE FUNCTION julia_setof_array()
RETURNS SETOF INTEGER[] AS $$
x =[[1,2], [3,4]]
for i in x
    return_next(i)
end
$$ LANGUAGE pljulia;
CREATE TYPE named_value AS (
  name   text,
  value  integer
);
CREATE FUNCTION julia_setof_composite()
RETURNS SETOF named_value AS $$
x = [("John", 1), ("Mary", 2)]
for i in x
    return_next(i)
end
$$ LANGUAGE pljulia;
SELECT julia_setof_int();
 julia_setof_int 
-----------------
               1
               2
               3
               4
               5
(5 rows)

SELECT julia_setof_array();
 julia_setof_array 
-------------------
 {1,2}
 {3,4}
(2 rows)

SELECT julia_setof_composite();
 julia_setof_composite 
-----------------------
 (John,1)
 (Mary,2)
(2 rows)

DROP FUNCTION julia_setof_array();
DROP FUNCTION julia_setof_int();
DROP FUNCTION julia_setof_composite();
DROP TYPE named_value;
